source/taskmanager/src/main/java/nl/aerius/taskmanager/mq/RabbitMQMessageHandler.java

```diff
@@ -147,24 +150,28 @@ private void delayRetry() {
     }
   }
 
-  private void stopAndStartConsumer() throws IOException {
+  private boolean stopAndStartConsumer() throws IOException {
     synchronized (this) {
       if (consumer != null) {
         consumer.stopConsuming();
       }
-      consumer = new RabbitMQMessageConsumer(
-          factory.getConnection().createChannel(),
-          queueConfig,
-          this);
-      consumer.getChannel().addShutdownListener(this::handleShutdownSignal);
-      consumer.startConsuming();
-      tryConnecting.set(false);
-      warned = false;
+      final Optional<Channel> channel = factory.getConnection().openChannel();
+
+      if (channel.isPresent()) {
+        consumer = new RabbitMQMessageConsumer(channel.get(), queueConfig, this);
+        consumer.getChannel().addShutdownListener(this::handleShutdownSignal);
+        consumer.startConsuming();
+        tryConnecting.set(false);
+        warned = false;
+        return true;
+      } else {
```

Comment:
So if it's not present, it'll just retry again without delay? Not sure what conditions that can be the case, or if that could be problematic.

Reply:
A right. yes it should delay. Moved the delay out of the catch. It should always delay if it is going to retry, independent if an exception was catched.